terraform/examples/io500/get_io500_result_data.sh

```diff
@@ -0,0 +1,203 @@
+# the run_io500-sc22.sh script has finished running.
+# Log out of the client and then in your cloud shell or local system
+# run ./get_io500_result_data.sh
+
+set -e
+trap 'echo "Hit an unexpected and unchecked error. Exiting."' ERR
+
+SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+
+ACTIVE_CONFIG="${SCRIPT_DIR}/config/active_config.sh"
+
+if [[ -L ${ACTIVE_CONFIG} ]]; then
+  # shellcheck source=/dev/null
+  source "$(readlink "${ACTIVE_CONFIG}")"
+else
+  log.error "No active config exists in ${ACTIVE_CONFIG}. Exiting..."
+  exit 1
+fi
+
+LOCAL_RESULTS_DIR="${SCRIPT_DIR}/results"
+LOCAL_RESULTS_DATA_DIR="${LOCAL_RESULTS_DIR}/io500_results"
+
+# BEGIN: Logging variables and functions
+LOG_LEVEL=INFO
+
+declare -A LOG_LEVELS=([DEBUG]=0 [INFO]=1  [WARN]=2   [ERROR]=3 [FATAL]=4 [OFF]=5)
+declare -A LOG_COLORS=([DEBUG]=2 [INFO]=12 [WARN]=3 [ERROR]=1 [FATAL]=9 [OFF]=0 [OTHER]=15)
+LOG_LINE_CHAR="-"
+if [[ -t 1 ]]; then
+  LOG_COLS=$(tput cols)
+else
+  LOG_COLS=40
+fi
+
+log() {
+  local msg="$1"
+  local lvl=${2:-INFO}
+  if [[ ${LOG_LEVELS[$LOG_LEVEL]} -le ${LOG_LEVELS[$lvl]} ]]; then
+    if [[ -t 1 ]]; then tput setaf "${LOG_COLORS[$lvl]}"; fi
+    printf "[%-5s] %s\n" "$lvl" "${msg}" 1>&2
+    if [[ -t 1 ]]; then tput sgr0; fi
+  fi
+}
+
+log.debug() { log "${1}" "DEBUG" ; }
+log.info()  { log "${1}" "INFO"  ; }
+log.warn()  { log "${1}" "WARN"  ; }
+log.error() { log "${1}" "ERROR" ; }
+log.fatal() { log "${1}" "FATAL" ; }
+
+log.line() {
+  local line_char="${1:-$LOG_LINE_CHAR}"
+  local line_width="${2:-$LOG_COLS}"
+  local fg_color="${3:-${LOG_COLORS[OTHER]}}"
+  local line
+  line=$(printf "%${line_width}s" | tr " " "${line_char}")
+  if [[ ${LOG_LEVELS[$LOG_LEVEL]} -le ${LOG_LEVELS[OFF]} ]]; then
+    if [[ -t 1 ]];then tput setaf "${fg_color}"; fi
+    printf -- "%s\n" "${line}" 1>&2
+    if [[ -t 1 ]]; then tput sgr0; fi
+  fi
+}
+
+# log.section msg [line_width] [line_char] [fg_color]
+log.section() {
+  local msg="${1:-}"
+  local line_width="${2:-$LOG_COLS}"
+  local line_char="${3:-$LOG_LINE_CHAR}"
+  local fg_color="${4:-${LOG_COLORS[OTHER]}}"
+  if [[ ${LOG_LEVELS[$LOG_LEVEL]} -le ${LOG_LEVELS[OFF]} ]]; then
+    log.line "${line_char}" "${line_width}" "${fg_color}"
+    if [[ -t 1 ]];then tput setaf "${fg_color}"; fi
+    printf "%s\n" "${msg}" 1>&2
+    log.line "${line_char}" "${line_width}" "${fg_color}"
+    if [[ -t 1 ]]; then tput sgr0; fi
+  fi
+}
+# END: Logging variables and functions
+
+get_results_tar_files() {
+  mkdir -p "${LOCAL_RESULTS_DATA_DIR}"
+  FIRST_CLIENT_IP=$(grep ssh "${SCRIPT_DIR}/login" | awk '{print $4}')
+  log.debug "FIRST_CLIENT_IP=${FIRST_CLIENT_IP}"
+  ssh -F ./tmp/ssh_config "${FIRST_CLIENT_IP}" "ls -1 ${IO500_TEST_CONFIG}*.tar.gz" > /tmp/results_tar_files
+
+  # shellcheck disable=SC2013
+  for f in $(cat /tmp/results_tar_files); do
+    LOCAL_RESULTS_TAR_FILE="${LOCAL_RESULTS_DATA_DIR}/${f}"
+    if [[ ! -f "${LOCAL_RESULTS_TAR_FILE}" ]]; then
```

Comment:
Could be usefull md5sum to also check if the files are differents and to overwrite the local one if needed ?

Reply:
Since the files include a timestamp in the name it would be very rare that the remote and local files would have the same name but different contents. The only way that would happen is if someone manually updated one of the tar.gz files after they were downloaded. Typically we run start.sh, get_io500_result_data.sh and then stop.sh. Most of the time we are just downloading a single file. Sometimes we do multiple runs and download 2 or three files. Since this script is just a helper for us to copy the results files locally before deleting the cluster, the work to add the md5 check doesn't seem necessary.